Add cookie support for fakku

Nodja opened the pull request:
This fixes #475.

I never used perl before so hopefully I got everything right.

Difegue commented on lib/LANraragi/Plugin/Login/Fakku.pm:
```diff
@@ -0,0 +1,60 @@
+package LANraragi::Plugin::Login::Fakku;
```
Please add this package to the list of tested modules in https://github.com/Difegue/LANraragi/blob/dev/tests/modules.t

Difegue commented on lib/LANraragi/Plugin/Login/Fakku.pm:
```diff
@@ -0,0 +1,60 @@
+
+    shift;
+    my ( $fakku_sid ) = @_;
+    return get_user_agent( $fakku_sid );
```
Not sure we really need a spare function here since there's not much going on.

Difegue commented on lib/LANraragi/Plugin/Login/Fakku.pm:
```diff
@@ -0,0 +1,60 @@
+        author    => "Nodja",
+        version   => "0.1",
+        description =>
+          "Handles login to fakku. The cookie is only valid for 7 days so don't forget to update it.",
```
Having to update the cookie every 7 days kinda sucks, but I'm guessing the login procedure is too annoying to automate.   
Is `fakku_sid` the only cookie that's needed? I don't actually have a Fakku account so I haven't tested this.

Nodja replied:
Yeah only fakku_sid is needed, the login form uses the a hidden google recaptcha field so I didn't even bother trying.

Difegue reviewed (changes requested):
Looks fine, I only have a few minor nitpicks. Thanks for tackling this!    
I renamed some stuff in the fakku plugin so you might have to rebase the PR, sorry 🙏

Nodja commented:
I added the module to the tests but had to reshuffle everything to keep the 2 column structure aligned.

The old variable names were funny reference but I agree it's probably not a good idea to keep that kind of naming.

Hopefully I rebased everything properly.

Difegue commented on lib/LANraragi/Plugin/Metadata/Fakku.pm:
```diff
@@ -24,7 +24,7 @@ sub plugin_info {
         type        => "metadata",
         namespace   => "fakkumetadata",
         author      => "Difegue",
```
Looks like the `loginfrom` field disappeared during the rebase here.

Nodja replied:
yep, I probably deleted it by mistake when resolving the conflicts

Difegue commented on lib/LANraragi/Plugin/Metadata/Fakku.pm:
```diff
@@ -24,7 +24,7 @@ sub plugin_info {
         author      => "Difegue",
-        version     => "0.5.1",
+        version     => "0.6",
         description => "Searches FAKKU for tags matching your archive.",
```
Since we now have cookie support, it might be worth it to edit the description to say you'll only get controversial content metadata if you login.  

 ```suggestion
        description => "Searches FAKKU for tags matching your archive. If you have an account, don't forget to enter the matching cookie in the login plugin to be able to access controversial content.",
```

Difegue commented on tests/modules.t:
```diff
@@ -26,34 +26,34 @@ require $cwd . "/tests/mocks.pl";
+    "LANraragi::Plugin::Login::EHentai",                    "LANraragi::Plugin::Login::Fakku",
+    "LANraragi::Plugin::Scripts::SourceFinder",             "LANraragi::Plugin::Scripts::FolderToCat",
+    "LANraragi::Plugin::Download::EHentai",                 "LANraragi::Plugin::Download::Chaika",
+    "LANraragi::Plugin::Scripts::nHentaiSourceConverter",   "LANraragi::Plugin::Scripts::Normalizer"
```
The testcount on line 7 needs to be bumped up to 56 😅 I always forget about this too.

Difegue commented:
All clear, I'll be merging this. Thanks again! 🙌
